aten/src/ATen/native/native_functions.yaml

```diff
@@ -6217,6 +6217,18 @@
     CPU: trace_cpu
     CUDA: trace_cuda
 
+- func: linalg_trace(Tensor self, int offset=0) -> Tensor
+  python_module: linalg
+  variants: method, function
+  dispatch:
+    CPU, CUDA: linalg_trace
```

Comment:
This line is overwritten by `CompositeExplicitAutograd: linalg_trace`. The code in the `linalg_trace` function is independent of the device so `CPU, CUDA` specialization is not needed here and `CompositeExplicitAutograd` is the correct choice of the dispatch key.
```suggestion

```

Reply:
On a second thought using `CompositeImplicitAutograd` should be better, then the backward function is not needed.